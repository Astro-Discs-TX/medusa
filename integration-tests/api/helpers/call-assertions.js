const { useApi } = require("../../helpers/use-api");

const header = {
  headers: {
    authorization: "Bearer test_token",
  },
};

const resolveCall = async (path, payload, header) => {
  const api = useApi();
  let res;
  try {
    res = (await api.post(path, payload, header)).status;
  } catch (e) {
    res = e.response.status;
  }
  return res;
};

const printIfFailing = (actual, expected, path) => {
  if (expected !== actual) {
    console.log(`failed at path : ${path}`);
  }
};

module.exports.expectPostCallToReturn = async function (
  input = {
    code,
    path,
    payload: {},
  }
) {
  const res = await resolveCall(input.path, input.payload, header);
  printIfFailing(res, input.code, input.path);
  expect(res).toEqual(input.code);
};

module.exports.expectAllPostCallsToReturn = async function (
  input = {
    code,
    col,
    pathf,
    payloadf,
  }
) {
  for (const i of input.col) {
    const res = await resolveCall(
      input.pathf(i),
      input.payloadf ? input.payloadf(i) : {},
      header
    );
    printIfFailing(res, input.code, input.pathf(i));
    expect(res).toEqual(input.code);
  }
};

module.exports.getObj = async function (path, obj) {
  const api = useApi();
  return (await api.get(path, header))?.data[obj];
};
