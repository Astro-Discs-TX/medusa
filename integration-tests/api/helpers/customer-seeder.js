const { Customer } = require("@medusajs/medusa");

module.exports = async (connection, data = {}) => {
  const manager = connection.manager;

  await manager.insert(Customer, {
    id: "test-customer-1",
    email: "test1@email.com",
  });

  await manager.insert(Customer, {
    id: "test-customer-2",
    email: "test2@email.com",
  });

  await manager.insert(Customer, {
    id: "test-customer-3",
    email: "test3@email.com",
  });
};
