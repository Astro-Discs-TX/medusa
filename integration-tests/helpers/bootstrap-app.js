const path = require("path")
const ExpressAdapter = require('medusa-platform-express');
const getPort = require("get-port")
const importFrom = require("import-from")

module.exports = {
  bootstrapApp: async ({ cwd } = {}) => {
    const httpAdapter = new ExpressAdapter();

    const loaders = importFrom(
      cwd || process.cwd(),
      "@medusajs/medusa/dist/loaders"
    ).default

    const { app, container, dbConnection } = await loaders({
      directory: path.resolve(cwd || process.cwd()),
      httpAdapter,
      isTest: false,
    })

    const PORT = await getPort()

    return {
      container,
      db: dbConnection,
      app,
      port: PORT,
    }
  },
}
