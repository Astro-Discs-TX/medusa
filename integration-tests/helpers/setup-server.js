const path = require("path")
const { spawn } = require("child_process")

const { setPort } = require("./use-api")

module.exports = ({ cwd, verbose }) => {
  const serverPath = path.join(__dirname, "test-server.js")

  return new Promise((resolve, reject) => {
    const workerId = parseInt(process.env.JEST_WORKER_ID || "1")
    const medusaProcess = spawn("node", [path.resolve(serverPath)], {
      cwd,
      env: {
        ...process.env,
        NODE_ENV: "development",
        WORKER_ID: workerId,
        JWT_SECRET: "test",
        COOKIE_SECRET: "test",
      },
      stdio: verbose
        ? ["inherit", "inherit", "inherit", "ipc"]
        : ["ignore", "ignore", "ignore", "ipc"],
    })

    medusaProcess.on("uncaughtException", (err) => {
      medusaProcess.kill()
    })

    medusaProcess.on("message", (port) => {
      setPort(port)
      resolve(medusaProcess)
    })
  })
}
