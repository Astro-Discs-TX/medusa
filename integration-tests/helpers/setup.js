import path from "path";
import express from "express";
import { createDatabase, dropDatabase } from "pg-god";

import loaders from "@medusajs/medusa/dist/loaders";

export const initialize = async (dir) => {
  const databaseName = "medusa-integration";

  await createDatabase({ databaseName });

  const app = express();

  const { dbConnection } = await loaders({
    directory: dir,
    expressApp: app,
  });

  const PORT = await getPort();

  return {
    db: dbConnection,
    app,
    port: PORT,
  };
};
