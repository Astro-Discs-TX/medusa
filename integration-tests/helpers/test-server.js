const path = require("path")
const express = require("express")
const getPort = require("get-port")
const importFrom = require("import-from")
const workerId = parseInt(process.env.JEST_WORKER_ID || "1")

const initialize = async () => {
  const configPath = path.resolve(path.join(process.cwd(), `medusa-config.js`))
  const configModule = require(configPath)

  configModule.projectConfig.database_url = `${configModule.projectConfig.database_url}-${workerId}`

  const app = express()

  const loaders = importFrom(
    process.cwd(),
    "@medusajs/medusa/dist/loaders"
  ).default

  const { dbConnection } = await loaders({
    directory: path.resolve(process.cwd()),
    expressApp: app,
    configModule,
  })

  const PORT = await getPort()

  return {
    db: dbConnection,
    app,
    port: PORT,
  }
}

const setup = async () => {
  const { app, port } = await initialize()

  app.listen(port, (err) => {
    process.send(port)
  })
}

setup()
