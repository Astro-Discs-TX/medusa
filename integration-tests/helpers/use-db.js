const { dropDatabase } = require("pg-god");

const DbTestUtil = {
  db_: null,

  setDb: function (connection) {
    this.db_ = connection;
  },

  clear: function () {
    return this.db_.synchronize(true);
  },

  shutdown: async function () {
    await this.db_.close();
    return dropDatabase({ databaseName });
  },
};

const instance = DbTestUtil;

module.exports = {
  setConnection: function (conn) {
    instance.setDb(conn);
    return instance;
  },
  useDb: function () {
    return instance;
  },
};
