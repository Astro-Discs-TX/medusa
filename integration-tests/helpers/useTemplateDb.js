const path = require("path")
require("dotenv").config({ path: path.join(__dirname, "../.env") })

const { createDatabase, dropDatabase } = require("pg-god")
const { createConnection, getConnection } = require("typeorm")

const DB_USERNAME = process.env.DB_USERNAME || "postgres"
const DB_PASSWORD = process.env.DB_PASSWORD || ""
const DB_URL = `postgres://${DB_USERNAME}:${DB_PASSWORD}@localhost`

const pgGodCredentials = {
  user: DB_USERNAME,
  password: DB_PASSWORD,
}

class TemplateDb {
  constructor() {
    this.connection_ = null
  }

  async createMasterConnection() {
    const connection = await createConnection({
      type: "postgres",
      name: "master",
      url: `${DB_URL}`,
    })

    return connection
  }

  async createDb_() {
    try {
      const connection = await this.getMasterConnection()

      const migrationDir = path.resolve(
        path.join(
          process.cwd(),
          "integration-tests",
          "api",
          `node_modules`,
          `@medusajs`,
          `medusa`,
          `dist`,
          `migrations`
        )
      )

      const templateDbName = "medusa-integration-template"

      await dropDatabase(
        {
          databaseName: templateDbName,
          errorIfNonExist: false,
        },
        pgGodCredentials
      )
      await createDatabase({ databaseName: templateDbName }, pgGodCredentials)

      const templateDbConnection = await createConnection({
        type: "postgres",
        name: "templateConnection",
        url: `${DB_URL}/${templateDbName}`,
        migrations: [`${migrationDir}/*.js`],
      })

      const res = await templateDbConnection.runMigrations()
      await templateDbConnection.close()

      return connection
    } catch (err) {
      console.log("createdb heeeeeeeeeeeeeeere")
      console.log(err)
    }
  }

  async getMasterConnection() {
    try {
      return await getConnection("master")
    } catch (err) {
      return await this.createMasterConnection()
    }
  }

  async createFromTemplate(dbName) {
    const connection = await this.getMasterConnection()

    await connection.query(`DROP DATABASE IF EXISTS "${dbName}";`)
    await connection.query(
      `CREATE DATABASE "${dbName}" TEMPLATE "medusa-integration-template";`
    )
  }

  async destroy() {
    let connection = await this.getMasterConnection()

    await connection.query(
      `DROP DATABASE IF EXISTS "medusa-integration-template";`
    )
    await connection.close()
  }
}

module.exports = new TemplateDb()
