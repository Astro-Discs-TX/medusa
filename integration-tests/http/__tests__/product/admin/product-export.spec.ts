import { IEventBusModuleService } from "@medusajs/types"
import { TestEventUtils, medusaIntegrationTestRunner } from "medusa-test-utils"
import {
  adminHeaders,
  createAdminUser,
} from "../../../../helpers/create-admin-user"
import { getProductFixture } from "../../../../helpers/fixtures"
import fs from "fs/promises"
import path from "path"
import { ModuleRegistrationName } from "@medusajs/utils"

jest.setTimeout(50000)

const compareCSVs = async (filePath, expectedFilePath) => {
  let fileContent = await fs.readFile(filePath, { encoding: "utf-8" })
  let fixturesContent = await fs.readFile(expectedFilePath, {
    encoding: "utf-8",
  })

  // Normalize csv data to get rid of dynamic data
  const idsToReplace = ["prod_", "pcol_", "variant_", "ptyp_"]
  const dateRegex =
    /(\d{4})-(\d{2})-(\d{2})T(\d{2})\:(\d{2})\:(\d{2})\.(\d{3})Z/g
  idsToReplace.forEach((prefix) => {
    fileContent = fileContent.replace(
      new RegExp(`${prefix}\\w*\\d*`, "g"),
      "<ID>"
    )
    fixturesContent = fixturesContent.replace(
      new RegExp(`${prefix}\\w*\\d*`, "g"),
      "<ID>"
    )
  })
  fileContent = fileContent.replace(dateRegex, "<DATE>")
  fixturesContent = fixturesContent.replace(dateRegex, "<DATE>")

  expect(fileContent).toEqual(fixturesContent)
}

medusaIntegrationTestRunner({
  testSuite: ({ dbConnection, getContainer, api }) => {
    let baseProduct
    let proposedProduct

    let baseCollection
    let publishedCollection

    let baseType

    let eventBus: IEventBusModuleService
    beforeAll(async () => {
      eventBus = getContainer().resolve(ModuleRegistrationName.EVENT_BUS)
    })

    beforeEach(async () => {
      await createAdminUser(dbConnection, adminHeaders, getContainer())

      baseCollection = (
        await api.post(
          "/admin/collections",
          { title: "base-collection" },
          adminHeaders
        )
      ).data.collection

      publishedCollection = (
        await api.post(
          "/admin/collections",
          { title: "proposed-collection" },
          adminHeaders
        )
      ).data.collection

      baseType = (
        await api.post(
          "/admin/product-types",
          { value: "test-type" },
          adminHeaders
        )
      ).data.product_type

      baseProduct = (
        await api.post(
          "/admin/products",
          getProductFixture({
            title: "Base product",
            description: "test-product-description\ntest line 2",
            collection_id: baseCollection.id,
            type_id: baseType.id,
            variants: [
              {
                title: "Test variant",
                prices: [
                  {
                    currency_code: "usd",
                    amount: 100,
                  },
                  {
                    currency_code: "eur",
                    amount: 45,
                  },
                  {
                    currency_code: "dkk",
                    amount: 30,
                  },
                ],
                options: {
                  size: "large",
                  color: "green",
                },
              },
              {
                title: "Test variant 2",
                prices: [
                  {
                    currency_code: "usd",
                    amount: 200,
                  },
                  {
                    currency_code: "eur",
                    amount: 65,
                  },
                  {
                    currency_code: "dkk",
                    amount: 50,
                  },
                ],
                options: {
                  size: "small",
                  color: "green",
                },
              },
            ],
          }),
          adminHeaders
        )
      ).data.product

      proposedProduct = (
        await api.post(
          "/admin/products",
          getProductFixture({
            title: "Proposed product",
            status: "proposed",
            tags: [{ value: "new-tag" }],
            type_id: baseType.id,
          }),
          adminHeaders
        )
      ).data.product
    })

    afterEach(() => {
      ;(eventBus as any).eventEmitter_.removeAllListeners()
    })

    describe("POST /admin/products/export", () => {
      it("should export a csv file containing the expected products", async () => {
        const subscriberExecution = TestEventUtils.waitSubscribersExecution(
          "notification.notification.created",
          eventBus
        )

        // BREAKING: The batch endpoints moved to the domain routes (admin/batch-jobs -> /admin/products/export). The payload and response changed as well.
        const batchJobRes = await api.post(
          "/admin/products/export",
          {},
          adminHeaders
        )

        const transactionId = batchJobRes.data.transaction_id
        expect(transactionId).toBeTruthy()

        await subscriberExecution
        const notifications = (
          await api.get("/admin/notifications", adminHeaders)
        ).data.notifications

        expect(notifications.length).toBe(1)
        expect(notifications[0]).toEqual(
          expect.objectContaining({
            data: expect.objectContaining({
              title: "Product export",
              description: "Product export completed successfully!",
              file: expect.objectContaining({
                url: expect.stringContaining("-product-exports.csv"),
                filename: expect.any(String),
                mimeType: "text/csv",
              }),
            }),
          })
        )

        await compareCSVs(
          notifications[0].data.file.url,
          path.join(__dirname, "__fixtures__", "exported-products.csv")
        )
        await fs.rm(path.dirname(notifications[0].data.file.url), {
          force: true,
          recursive: true,
        })
      })

      it("should export a csv file filtered by specific products", async () => {
        const subscriberExecution = TestEventUtils.waitSubscribersExecution(
          "notification.notification.created",
          eventBus
        )

        // BREAKING: We don't support setting batch size in the export anymore
        const batchJobRes = await api.post(
          `/admin/products/export?id=${proposedProduct.id}`,
          {},
          adminHeaders
        )

        const transactionId = batchJobRes.data.transaction_id
        expect(transactionId).toBeTruthy()

        await subscriberExecution
        const notifications = (
          await api.get("/admin/notifications", adminHeaders)
        ).data.notifications

        expect(notifications.length).toBe(1)

        await compareCSVs(
          notifications[0].data.file.url,
          path.join(__dirname, "__fixtures__", "filtered-products.csv")
        )

        await fs.rm(path.dirname(notifications[0].data.file.url), {
          force: true,
          recursive: true,
        })
      })
    })
  },
})
