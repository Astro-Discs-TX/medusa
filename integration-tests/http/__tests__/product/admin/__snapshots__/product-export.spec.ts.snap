// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` POST /admin/products/export should export a csv file containing the expected products 1`] = `
[
  {
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description
test line 2",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "base-product",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "draft",
    "Product Subtitle": "",
    "Product Tag 1": "tag-123",
    "Product Tag 2": "tag-456",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Base product",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "large",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price DKK": 30,
    "Variant Price EUR": 45,
    "Variant Price USD": 100,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
  {
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description
test line 2",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "base-product",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "draft",
    "Product Subtitle": "",
    "Product Tag 1": "tag-123",
    "Product Tag 2": "tag-456",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Base product",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "small",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price DKK": 50,
    "Variant Price EUR": 65,
    "Variant Price USD": 200,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant 2",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
  {
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "proposed-product",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "proposed",
    "Product Subtitle": "",
    "Product Tag 1": "new-tag",
    "Product Tag 2": "",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Proposed product",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "large",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price DKK": 30,
    "Variant Price EUR": 45,
    "Variant Price USD": 100,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
]
`;

exports[` POST /admin/products/export should export a csv file filtered by specific products 1`] = `
[
  {
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "proposed-product",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "proposed",
    "Product Subtitle": "",
    "Product Tag 1": "new-tag",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Proposed product",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "large",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price DKK": 30,
    "Variant Price EUR": 45,
    "Variant Price USD": 100,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
]
`;

exports[` POST /admin/products/export should export a csv file with categories 1`] = `
[
  {
    "Product Category 1": "<ID>",
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description
test line 2",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "base-product",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "draft",
    "Product Subtitle": "",
    "Product Tag 1": "tag-123",
    "Product Tag 2": "tag-456",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Base product",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "large",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price DKK": 30,
    "Variant Price EUR": 45,
    "Variant Price USD": 100,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
  {
    "Product Category 1": "<ID>",
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description
test line 2",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "base-product",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "draft",
    "Product Subtitle": "",
    "Product Tag 1": "tag-123",
    "Product Tag 2": "tag-456",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Base product",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "small",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price DKK": 50,
    "Variant Price EUR": 65,
    "Variant Price USD": 200,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant 2",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
]
`;

exports[` POST /admin/products/export should export a csv file with region prices 1`] = `
[
  {
    "Product Collection Id": "<ID>",
    "Product Created At": "<DateTime>",
    "Product Deleted At": "<DateTime>",
    "Product Description": "test-product-description",
    "Product Discountable": true,
    "Product External Id": "<ID>",
    "Product Handle": "product-with-prices",
    "Product Height": "",
    "Product Hs Code": "",
    "Product Id": "<ID>",
    "Product Image 1": "test-image.png",
    "Product Image 2": "test-image-2.png",
    "Product Is Giftcard": false,
    "Product Length": "",
    "Product Material": "",
    "Product Mid Code": "",
    "Product Origin Country": "",
    "Product Status": "draft",
    "Product Subtitle": "",
    "Product Tag 1": "tag-123",
    "Product Tag 2": "tag-456",
    "Product Thumbnail": "test-image.png",
    "Product Title": "Product with prices",
    "Product Type Id": "<ID>",
    "Product Updated At": "<DateTime>",
    "Product Weight": "",
    "Product Width": "",
    "Variant Allow Backorder": false,
    "Variant Barcode": "",
    "Variant Created At": "<DateTime>",
    "Variant Deleted At": "<DateTime>",
    "Variant Ean": "",
    "Variant Height": "",
    "Variant Hs Code": "",
    "Variant Id": "<ID>",
    "Variant Length": "",
    "Variant Manage Inventory": true,
    "Variant Material": "",
    "Variant Metadata": "",
    "Variant Mid Code": "",
    "Variant Option 1 Name": "size",
    "Variant Option 1 Value": "large",
    "Variant Option 2 Name": "color",
    "Variant Option 2 Value": "green",
    "Variant Origin Country": "",
    "Variant Price Test Region [USD]": 45,
    "Variant Price USD": 100,
    "Variant Product Id": "<ID>",
    "Variant Sku": "",
    "Variant Title": "Test variant",
    "Variant Upc": "",
    "Variant Updated At": "<DateTime>",
    "Variant Variant Rank": 0,
    "Variant Weight": "",
    "Variant Width": "",
  },
]
`;
